File: nodirnocry.js
const { Command } = require('commander');
const axios = require('axios');
const chalk = require('chalk');
const fs = require('fs');
const path = require('path');

// Utility to pause execution
const sleep = (ms) => new Promise(resolve => setTimeout(resolve, ms));

const program = new Command();

program
  .option('-u, --url <urls>', 'target URL(s)', val => val.split(','))
  .option('-w, --wordlist <file>', 'path to wordlist')
  .option('-d, --delay <ms>', 'delay between requests in milliseconds', parseInt)
  .parse();

const options = program.opts();

function formatUrl(url) {
  if (!/^https?:\/\//i.test(url)) {
    url = 'http://' + url;
  }
  return url;
}

async function checkURL(baseURL, word) {
  const formattedURL = formatUrl(baseURL);
  try {
    const response = await axios.get(`${formattedURL}/${word}`, {
      maxRedirects: 5,
      validateStatus: (status) => status >= 200 && status < 400,
    });

    if (response.status >= 200 && response.status < 400) {
      console.log(chalk.green(`Found: ${formattedURL}/${word} (Status: ${response.status})`));
    } else {
      console.log(chalk.yellow(`Not Found: ${formattedURL}/${word} (Status: ${response.status})`));
    }
  } catch (error) {
    if (error.response) {
      console.log(chalk.red(`Error: ${formattedURL}/${word} - Status: ${error.response.status}`));
    } else {
      console.log(chalk.red(`Error: ${formattedURL}/${word} - Network issue or timeout`));
    }
  }
}

async function startScan(urls, wordlist, delay = 0) {
  for (const url of urls) {
    console.log(chalk.blue(`Starting scan on: ${url}`));
    const words = fs.readFileSync(wordlist, 'utf-8')
      .split('\n')
      .map(line => line.trim())
      .filter(line => line.length > 0);

    if (words.length === 0) {
      console.log(chalk.yellow('Warning: Wordlist is empty.'));
      return;
    }

    for (const word of words) {
      await checkURL(url, word);
      if (delay) await sleep(delay);
    }
  }
}

if (!options.url || !options.wordlist) {
  console.log(chalk.red('Error: URL(s) and wordlist are required.'));
  process.exit(1);
}

const scanDelay = options.delay || 0;
startScan(options.url, options.wordlist, scanDelay);

File: nodirnocry.py
import requests
import argparse
import time
from concurrent.futures import ThreadPoolExecutor
from colorama import Fore, Style, init
from pyfiglet import Figlet


init(autoreset=True)

def print_banner():
    f = Figlet(font='slant')  # You can try 'standard', 'slant', 'block', etc.
    print(Fore.MAGENTA + f.renderText("NoDirNoCry") + Style.RESET_ALL)


def format_url(url):
    if not url.startswith(('http://', 'https://')):
        return 'http://' + url
    return url

def check_url(base_url, path):
    full_url = f"{base_url.rstrip('/')}/{path.strip()}"
    try:
        response = requests.get(full_url, timeout=5, allow_redirects=True)
        if response.status_code < 400:
            print(f"{Fore.GREEN}[+] Found: {full_url} [{response.status_code}]")
        else:
            print(f"{Fore.YELLOW}[-] Not Found: {full_url} [{response.status_code}]")
    except requests.RequestException as e:
        print(f"{Fore.RED}[!] Error: {full_url} ({e})")

def main():
    print_banner()  # <- Print ASCII banner first
    parser = argparse.ArgumentParser(description="NoDirNoCry 🕵️ - A simple Python-based directory scanner.")
    parser.add_argument("-u", "--url", required=True, help="Target URL")
    parser.add_argument("-w", "--wordlist", required=True, help="Path to wordlist")
    parser.add_argument("-c", "--concurrency", type=int, default=10, help="Number of concurrent threads")
    parser.add_argument("--delay", type=float, default=0.0, help="Delay between requests in seconds")

    args = parser.parse_args()

    base_url = format_url(args.url)

    try:
        with open(args.wordlist, "r") as file:
            words = [line.strip() for line in file.readlines() if line.strip()]
    except FileNotFoundError:
        print(f"{Fore.RED}[!] Wordlist file not found.")
        return

    print(f"{Fore.CYAN}[*] Starting scan on {base_url} with {args.concurrency} threads...")

    with ThreadPoolExecutor(max_workers=args.concurrency) as executor:
        for word in words:
            executor.submit(check_url, base_url, word)
            if args.delay:
                time.sleep(args.delay)

if __name__ == "__main__":
    main()
